node-red-project/flows.json

```diff
@@ -9372,5 +9380,521 @@
+            }
+        ],
+        "action": "",
+        "property": "",
+        "from": "",
+        "to": "",
+        "reg": false,
+        "x": 470,
+        "y": 260,
+        "wires": [
+            [
+                "c497cfedf6aed533"
+            ]
+        ]
+    },
+    {
+        "id": "b6b8a1a7c08eb0ac",
+        "type": "change",
+        "z": "f410f6aac9d1a3fc",
+        "name": "Loadavg",
+        "rules": [
+            {
+                "t": "set",
+                "p": "topic",
+                "pt": "msg",
+                "to": "metrics_Loadavg",
+                "tot": "str"
+            }
+        ],
+        "action": "",
+        "property": "",
+        "from": "",
+        "to": "",
+        "reg": false,
+        "x": 480,
+        "y": 320,
+        "wires": [
+            [
+                "c497cfedf6aed533"
+            ]
+        ]
+    },
+    {
+        "id": "976ec527b724bd3d",
+        "type": "change",
+        "z": "f410f6aac9d1a3fc",
+        "name": "Memory",
+        "rules": [
+            {
+                "t": "set",
+                "p": "topic",
+                "pt": "msg",
+                "to": "metrics_Memory",
+                "tot": "str"
+            }
+        ],
+        "action": "",
+        "property": "",
+        "from": "",
+        "to": "",
+        "reg": false,
+        "x": 480,
+        "y": 380,
+        "wires": [
+            [
+                "c497cfedf6aed533"
+            ]
+        ]
+    },
+    {
+        "id": "6881f1dd8a4805af",
+        "type": "change",
+        "z": "f410f6aac9d1a3fc",
+        "name": "NetworkIntf",
+        "rules": [
+            {
+                "t": "set",
+                "p": "topic",
+                "pt": "msg",
+                "to": "metrics_NetworkIntf",
+                "tot": "str"
+            }
+        ],
+        "action": "",
+        "property": "",
+        "from": "",
+        "to": "",
+        "reg": false,
+        "x": 490,
+        "y": 440,
+        "wires": [
+            [
+                "c497cfedf6aed533"
+            ]
+        ]
+    },
+    {
+        "id": "f894a3a8017a7de4",
+        "type": "function",
+        "z": "f410f6aac9d1a3fc",
+        "name": "set up tb metrics post",
+        "func": "// The server URL, port, and device access token must be stored in the config context\n// before this will work\nlet serverUrl = global.get(\"serverUrl\", \"config\")\nlet serverPort = global.get(\"serverPort\", \"config\")\nlet accessToken = global.get(\"accessToken\", \"config\")\nmsg.url = `${serverUrl}:${serverPort}/api/v1/${accessToken}/telemetry`\nmsg.method = \"POST\";\nreturn msg;",
+        "outputs": 1,
+        "timeout": 0,
+        "noerr": 0,
+        "initialize": "",
+        "finalize": "",
+        "libs": [],
+        "x": 500,
+        "y": 600,
+        "wires": [
+            [
+                "634f581fd150b60c"
+            ]
+        ]
+    },
+    {
+        "id": "521115ddd5741eea",
```

Comment:
Since all the system metrics are being queried and sent at the same time, let's bundle it as a single payload. i.e. Nest them under a single key, i.e. `system_metrics`:

```
// format the payload with a timestamp
let values = {
    ...msg.payload,
}
//ts in milliseconds
msg.payload = {
    ts: Date.now(),
    values: {
        system_metrics: values
    }
}

return msg;
```

Reply:
I wish I'd seen this before so I don't have to go back and make yet another change to this PR.

It seems like a cosmetic change?  Is it worth even 10 minutes of effort?

Currently, each type of metric is posted as an individual telemetry value, as in "metrics_CPUx", "metrics_Drives", etc.  This will "push down" the values, making it that much harder to decompose them.

Reply:
Having the system_metrics object around this allows us to change (add/remove) system metrics in the future without having to configure for the additional payloads outside of composing it.  This will be particularly useful downstream of ThingsBoard.

I made the changes on the [latest commit](https://github.com/glossyio/traffic-monitor/pull/24/commits/1e0b69a8d741954db2d54fa603cac096aef06bbd).